src/packet/BinaryWriter.js

```diff
@@ -1,137 +1,62 @@
-﻿'use strict';
+'use strict';
 /*
  * Simple BinaryWriter is a minimal tool to write binary stream with unpredictable size.
  * Useful for binary serialization.
  *
- * Copyright (c) 2016 Barbosik
- * 
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- * 
- *     http://www.apache.org/licenses/LICENSE-2.0
- * 
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- * 
- * 
+ * Copyright (c) 2016 Barbosik https://github.com/Barbosik
+ * License: Apache License, Version 2.0
  */
 
-function BinaryWriter(size) {
-    if (!size || size <= 0) {
-        size = Buffer.poolSize / 2;
-    }
-    this._buffer = new Buffer(size);
+function BinaryWriter() {
+    this._buffer = parseInt(process.version[1]) < 6 ? new Buffer(1e5) : Buffer.allocUnsafe(1e5);

```

Comment:
@Andrews54757 you're taking these bottlenecks very far. I'm running the very old MultiOgar on a server for just $35/m and it can handle around 80 players on a single node.js server (on very weak cores) just fine. You're taking all the bottlenecks a server might have to a point where it's just a waste of time to even consider adding the code for it.

Who with the right mind is going to get 4000 players onto 1 server? The server itself won't even be able to handle that much latency/load to work with it. I have experience from running on many different servers and working with Linux myself. Ogar IS NOT an enterprise software to be running on double-grade end hardware. It's meant for people to play around with or host their own servers, which it is perfectly fine and capable of doing right now.

Remember, it's an open source software, not a competition. You and your stupid licensing system is possibly the dumbest open source software method I have seen. You're closing down parts of the code that many programmers might want to see and remove if they don't want you tracking everything they do with your server.

OpenAgar is **not** open-source. I would even not disclose it as semi, if you don't want to provide help to people that don't know how to use it, then don't. Simply close the issue or disable issues completely if you find helping people a pain.